File: InterviewPrep1/LinkedList/firstLinkedlist.java
class Node {
     public int data;
     public Node next;

     Node()
     {
         this.data = 0;
         this.next = null;
     }

     Node(int data)
     {
         this.data = data;
         this.next = null;
     }

     Node(int data, Node next)
     {
         this.data = data;
         this.next = next;
     }
 };
public class firstLinkedlist {
    public static Node constructLL(int []arr) {
        // Write your code here
        Node head = new Node(arr[0]);
        Node nexthead = head;
        for(int i = 1; i < arr.length; i++){
            Node temp = new Node(arr[i]);
            nexthead.next = temp;
            nexthead = temp;
        }
        return head;
    }
    public static void main(String[] args) {
        int[] arr = {4,2,1,5};
        System.out.println(constructLL(arr));
    }
}

File: InterviewPrep1/LinkedList/firstLinkedlist.py
class Node:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def constructLL(arr: [int]) -> Node:
    # Write your code here
    if not arr:
        return None
    head = cur = Node(val=arr[0])
    for element in arr[1:]:
        cur.next = Node(val=element)
        cur = cur.next
    return head

arr = [4, 2, 1, 5]
linked_list_head = constructLL(arr)
# Printing the linked list elements
current_node = linked_list_head
while current_node:
    print(current_node.val, end=" -> ")
    current_node = current_node.next
print("None")